### ISSUE TITLE
Lookups stop working after 10 minutes

### ISSUE THREAD

**Issue by TravisLlado** [microsoft/vscode-cpptools#8674]
Issue Type: <b>Bug</b>

Lookups (Go to Definition, Go to Declaration, onMouseOver descriptions, etc) work as expected for the first ~10 minutes after I start VSCode, and suddenly they completely stop working. The "Loading ..." popup appears, the "working" animation runs across the top of the screen, but nothing ever happens. If I completely shutdown VSCode and immediately restart it, C/C++ lookups work for another ~10 minutes then stop again. C/C++ extension has been behaving like this for the past several months, since the initial release of 1.7.0.

Extension version: 1.7.1
VS Code version: Code - Insiders 1.64.0-insider (Universal) (c6e58d126f6b5676242e32b34180b2a595f29b51, 2022-01-14T05:16:38.175Z)
OS version: Darwin x64 21.1.0
Restricted Mode: No
Remote OS version: Linux x64 4.4.0-213-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (2.17GB free)|
|Process Argv|--crash-reporter-id 0157a282-35a0-4928-be2f-0864763e35ca|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: 10.65.110.188|
|OS|Linux x64 4.4.0-213-generic|
|CPUs|Intel(R) Xeon(R) Gold 6226R CPU @ 2.90GHz (4 x 2899)|
|Memory (System)|62.92GB (39.21GB free)|
|VM|100%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30422396
pythontb:30258533
vsc_aa:30263845
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805:30301674
pythonvspyt200:30323110
vsccppwt:30421239
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30404738
wslgetstartedc:30421358
vscop804:30404766
vscop940:30404999
vsclayoutctrt:30405800
azactmsalcf:30409288
vsccppwtstudy:30421978

```

</details>

**Comment by sean-mcmanus** [comment]
Do you know if this repros with https://github.com/microsoft/vscode-cpptools/releases/tag/1.8.0-insiders2 ?

Is CPU being used by the cpptools or cpptools-srv processes when this occurs?

Is CPU being used in the 10 minute period if you don't do any mouse interactions, i.e. is there background work occurring? Does the repro occur if you idle for 10 minutes and then start working or does it require 10 minutes of mouse interactions or edits or what?

When this repros, do you see any logging messages with C_Cpp.loggingLevel set to "Debug" before and/or during the repro?

**Comment by sean-mcmanus** [comment]
In particular, https://github.com/microsoft/vscode-cpptools/releases/tag/1.8.0-insiders2 has 2 known deadlock fixes.

Do you have configurationProvider set? If so, our upcoming 1.8.0-insiders3 has another potential deadlock fix with configurationProviders.

**Comment by TravisLlado** [comment]
Do you know if this repros with https://github.com/microsoft/vscode-cpptools/releases/tag/1.8.0-insiders2 ?
    I haven't tried 1.8.0 recently. I can check.

Is CPU being used by the cpptools or cpptools-srv processes when this occurs?
    Yes. When I run htop I see five identical processes
    /home/tllado/.vscode-server-insiders/extensions/ms-vscode.cpptools-1.7.1/bin/cpptools
    and each of them is consuming 100% of a CPU core.
    When I reboot my computer, I see two processes consuming 50% each.

Is CPU being used in the 10 minute period if you don't do any mouse interactions, i.e. is there background work occurring? Does the repro occur if you idle for 10 minutes and then start working or does it require 10 minutes of mouse interactions or edits or what?
    If I reboot my computer, when I start VSCode I see two cpptools processes, both consuming 50% of a CPU, and lookup functions work as expected. Lookups stop working 10 minutes after startup, regardless of user interaction.

Also, how consistent is the 10 minute repro? Does it seem like a random deadlock that takes around 10 minutes of usage?
    I haven't timed it exactly, but it seems like it's quite consistent, always ~10 minutes.

When this repros, do you see any logging messages with C_Cpp.loggingLevel set to "Debug" before and/or during the repro?
    Several hundred rows of what appears to be typical debug statements ... (filenames redacted)
        Error squiggle count: 1
        Update IntelliSense time (sec): 0.329
        cpptools/getSemanticTokens: file:///home/tllado/filename.h (id: 31)
        cpptools/getSemanticTokens: file:///home/tllado/filename.cpp (id: 32)
        cpptools/getFoldingRanges: file:///home/tllado/filename.h (id: 33)
        cpptools/getCodeActions: file:///home/tllado/filename.h (id: 34)
        textDocument/hover: file:///home/tllado/filename.h (id: 35)
    followed by 40,000 rows of
        tag parsing file: /home/tllado/.cache/bazel/_bazel_tllado/621b8961867ef688406c926eb935ac2c/execroot/...

Do you know if this only repros with a certain project or does it repro with any simple project?
    I've only see this happen on one computer with one particular project, but it happens 100% of the time on that project.

Also, do you have the same workspace open in multiple VS Code instances?
    No, I only have one computer accessing this workspace.

If I add
    "files.exclude": {
        "**/.cache": true,
    },
to project/.vscode/settings.json, then the problem appears to be solved and the lookup features continue to work. I don't get piles of "tag parsing file ..." messages.

**Comment by sean-mcmanus** [comment]
When you reboot your machine, you should have 0 cpptools processes. I assume you mean you have 2 when starting VS Code on the remote machine. htop is actually showing threads and not processes and our cpptools process creates a lot of processes, so for figuring out how many processes are running you should use "top" instead (or maybe there's some way to configure "htop" to get the actual processes). We only launch 1 cpptools process at a time, so on a fresh launch you shouldn't actually have 2 processes. I'm assuming there's only 1 process involved.

The 50% CPU usage continuously for 10 minutes suggests it is tag parsing in the background and then after 10 minutes it could be infinite looping when it tries to tag parse a particular file, which could be a duplicate of the fixed (in 1.8.0-insiders2) issue https://github.com/microsoft/vscode-cpptools/issues/8459 .

Yes, adding the files.exclude is the recommended "fix", of either the .cache folder or whatever subfolder is problematic.

Any folders that have code that you don't care about getting workspace symbol information for should be added to files.exclude or C_Cpp.files.exclude (if you still want to see the folder in the VS Code explorer).

Can I close this issue?

**Comment by TravisLlado** [comment]
1.8.0-2 seems to be working without adding the exemption to settings. We can close this ticket. Thank you.
